### PR TITLE
[INFRA/VOTING] Patch to Neaten Voting Implementation

### PR DESCRIPTION
**Pull request by preetl**
# Summary

Instead of creating alive client IDs in voting, they are passed in from IIGO.

EDIT: the patch got big

## Additional Information

N/A

## Test Plan

Still pass!

EDIT: added tests!

### CODE REVIEW

**Comment by QFSW** [comment]
# ⚠️Warning! This diff contains changes in protected folders

File: internal/server/iigointernal/judiciary.go [hunk 1]
```diff
@@ -133,7 +133,7 @@ func (j *judiciary) appointNextPresident(monitoring shared.MonitorResult, curren
 			return j.gameState.PresidentID, errors.Errorf("Insufficient Budget in common Pool: appointNextPresident")
 		}
 		election.ProposeElection(shared.President, electionsettings.VotingMethod)
-		election.OpenBallot(electionsettings.IslandsToVote, iigoClients)
+		election.OpenBallot(electionsettings.IslandsToVote, iigoClients, allIslands)
```
Comment (lhl2617) [inline internal/server/iigointernal/judiciary.go]: just an extra check that only non-dead islands are passed in, right?

Comment (preetl) [reply internal/server/iigointernal/judiciary.go]: allIslands is passed in from orchestration where it is populated with only alive client IDs

**Review by Ning-72 — APPROVED** [review]

**Review by NGovani — APPROVED** [review]
LGTM

**Pull request merged**